### ISSUE TITLE
[Training] How can I obtain the gradients of training parameters through the onnxruntime training C++API？

### ISSUE THREAD

**Issue by zhaowujie** [microsoft/onnxruntime#20231]
### Describe the issue

Thank you for allowing me to use onnxruntime code.
Is it possible to obtain gradients in the training loop through the onnxruntime training C++API?

### To reproduce

My code just like the sample code: [https://onnxruntime.ai/docs/api/c/training_c_cpp_api.html](https://onnxruntime.ai/docs/api/c/training_c_cpp_api.html)

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.17.1

### PyTorch Version

2.0.0

### Execution Provider

Default CPU

**Comment by xadupre** [comment]
Do you need the gradient in C++ or a python would be ok?

+@baijumeswani

**Comment by zhaowujie** [comment]
Thank you for your reply.

> Do you need the gradient in C++ or a python would be ok?

Due to machine limitations, I need to use the C++interface to obtain gradients. Is there any sample code available?

**Comment by baijumeswani** [comment]
> Is it possible to obtain gradients in the training loop through the onnxruntime training C++API?

As of now, we have not provided a way to query the gradients from C/C++. May I ask what is the scenario where querying the gradients is needed?

**Comment by zhaowujie** [comment]
Thanks a lot~
I just want to obtain the Hessian matrix of weights through gradients.
